lucene/core/src/test/org/apache/lucene/index/TestIndexWriterMergePolicy.java

```diff
@@ -396,10 +396,12 @@ protected void merge(MergePolicy.OneMerge merge) throws IOException {
             waitForMerge.await();
             writer.updateDocument(new Term("id", "2"), d2);
             writer.flush();
-            waitForUpdate.countDown();
           } catch (Exception e) {
             throw new AssertionError(e);
+          } finally {
+            waitForUpdate.countDown();
```

Comment:
Ahh is this so the test does not hang on an (unexpected) exception in the `try` block?

Reply:
yes